test/e2e/ic_conditions_metric_test.go

```diff
@@ -0,0 +1,159 @@
+		if podsList, err = kubeClient.CoreV1().Pods("openshift-ingress-operator").List(context, metav1.ListOptions{}); err != nil || len(podsList.Items) == 0 {
+			return false, err
+		}
+		//We are checking that the new pod is different from the previous one
+		_, newOperatorPodUID := extractIngressOperatorPodNameAndUID(podsList)
+		if newOperatorPodUID == operatorPodUID {
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		return err
+	}
+	return nil
+}
+
+// extractIngressOperatorPodNameAndUID helper function to extract from a pods list the name and UID of the ingress operator pod
+func extractIngressOperatorPodNameAndUID(podsList *corev1.PodList) (string, types.UID) {
+	operatorPodName := ""
+	operatorPodUID := types.UID("")
+	for _, pod := range podsList.Items {
+		if strings.HasPrefix(pod.Name, "ingress-operator") {
+			operatorPodName = pod.Name
+			operatorPodUID = pod.UID
+			break
+		}
+	}
+	return operatorPodName, operatorPodUID
+}
+
+// TestIngressControllerConditionsMetricAfterRestart verifies that metric ingress_controller_conditions(router,status) is
+// available after an operator pod restart too.
+//
+// This test:
+//
+// 1. Verifies that the metric is available in a normal situation when the operator pod is up&running (i.e. before restart)
+//
+// 2. Restarts the operator pod, waiting it will be available again
+//
+// 3. Repeats the step 1 expecting the same result again and so the presence of the metric
+//
+// NB:
+//  1. this test requires an OpenShift version with the monitoring stack up&running
+//  2. due to the fact that this test is restarting the operator pod it cannot be executed in parallel with other tests
+func TestIngressControllerConditionsMetricAfterRestart(t *testing.T) {
+
+	// Create a new prometheus client for fetching metrics and dependencies needed
+	kubeConfig, err := config.GetConfig()
+	if err != nil {
+		t.Fatalf("Failed to get kube config: %s", err)
+	}
+	kubeClient, err := kubernetes.NewForConfig(kubeConfig)
+	if err != nil {
+		t.Fatalf("Failed to create kube client: %v", err)
+	}
+	routeClient, err := routev1client.NewForConfig(kubeConfig)
+	if err != nil {
+		t.Fatalf("Failed to create route client: %v", err)
+	}
+	prometheusClient, err := metrics.NewPrometheusClient(context.Background(), kubeClient, routeClient)
+	if err != nil {
+		t.Fatalf("Failed to create prometheus client: %v", err)
+	}
+
+	// Check metric before restart
+	t.Log("Verifying that in Prometheus metrics there are ingress_controller_conditions metrics before restart")
+	// Wait for metrics to be added and set to 0.
+	if err := waitForIngressControllerConditionsMetrics(t, prometheusClient); err != nil {
+		t.Fatalf("Failed to fetch expected metrics: %v", err)
+	}
+
+	// Restart operator pod
+	err = restartOperatorPod(t, kubeClient)
+	if err != nil {
+		t.Fatalf("Failed to restart operator pod: %v", err)
+	}
```

Comment:
A little more go idiomatic:
```suggestion
	if err := restartOperatorPod(t, kubeClient); err != nil {
		t.Fatalf("Failed to restart operator pod: %v", err)
	}
```